### DISCUSSION TITLE
reset() using null-values

### DISCUSSION THREAD

**Discussion by Fyers** [react-hook-form/react-hook-form#8603] [Q&A]
Updating from v6 to v7 I encountered a breaking change which I didnt find in the Migration-Guide.
Setting defaultValues, or using reset did not actually set the value of the field, if the value provided was null.

So e.g. doing `reset({ example: "test123", exampleTwo: null });` in v6 the value of `exampleTwo` in submit was still `""`. Since v7 it is `null`. Thats a problem for me in a specific use case where the value is sent via formValues. Since formValues are all strings the server gets `null`  as a string (`"null"`).

Is there any way to change this behaviour of RHF without making my own logic to remove null-values before resetting an form?

### v6
https://codesandbox.io/s/rhf-v6-null-value-es9jt5?file=/src/index.js
`{example: "test123", exampleTwo: ""}`

### v7
https://codesandbox.io/s/rhf-v7-null-value-0c3u79?file=/src/index.js
`{example: "test123", exampleTwo: null}`

**Comment by bluebill1049** [comment]
Could you set it as an empty string instead?

**Reply by Fyers** [reply]
That is one workaround. The problem is that since the data is coming from the server I need to change every model to use an empty string as default.

If there is no other way I would do something similar, but I rather had an workaround with some kind of setting if one exist.
